check-aws-cloudwatch-logs/lib/check-aws-cloudwatch-logs.go

```diff
@@ -107,66 +107,58 @@ type logState struct {
 	StartTime *int64
 }
 
-func (p *awsCloudwatchLogsPlugin) collect() ([]string, error) {
-	var nextToken *string
-	var startTime *int64
-	if s, err := p.loadState(); err != nil {
-		if !os.IsNotExist(err) {
-			return nil, err
-		}
-	} else {
-		if s.StartTime != nil && *s.StartTime > time.Now().Add(-time.Hour).Unix()*1000 {
-			nextToken = s.NextToken
-			startTime = s.StartTime
-		}
+func (p *awsCloudwatchLogsPlugin) collect(now time.Time) ([]string, error) {
+	s, err := p.loadState()
+	if err != nil {
+		return nil, err
 	}
-	if startTime == nil {
-		startTime = aws.Int64(time.Now().Add(-1*time.Minute).Unix() * 1000)
+	if s.StartTime == nil || *s.StartTime <= now.Add(-time.Hour).Unix()*1000 {
+		s.StartTime = aws.Int64(now.Add(-1*time.Minute).Unix() * 1000)
```

Comment:
Maybe we should set `s.NextToken = nil` explicitly for this case?

Reply:
You're right. I'll fix it!